esmvalcore/_config/_config.py

```diff
@@ -13,6 +16,46 @@
 
 CFG = {}
 
+if sys.version_info[:2] >= (3, 9):
+    # pylint: disable=no-name-in-module
+    from importlib.resources import files as importlib_files
+else:
+    from importlib_resources import files as importlib_files
+
+
+def _deep_update(dictionary, update):
+    for key, value in update.items():
+        if isinstance(value, collections.abc.Mapping):
+            dictionary[key] = _deep_update(dictionary.get(key, {}), value)
+        else:
+            dictionary[key] = value
+    return dictionary
+
+
+@lru_cache
+def _get_variable_details_for_project(project):
```

Comment:
Maybe you could rename the function to `_get_variable_details` because `project` is already the name of the argument, so it's obvious that it's `for_project`. Or maybe a name like `_load_extra_facets`? Or `_load_project_facets`?